### DISCUSSION TITLE
Batcup Upscaling

### DISCUSSION THREAD

**Discussion by omgbaby** [lllyasviel/Fooocus#3582] [Ideas]
First and foremost, thank you so much for creating fooocus. Your creation has changed my life for the better when I needed the most.

I am not a dev but I'm just curious; would there be a batch upscaling option in the Upscale or Variation be possible/feasible to develop? Just thought it'd be great to have that rather than needing to it manually one by one. It's not a big deal when it's a couple dozen, but I'm working with thousands these days for production and it's really not a joke for me anymore lol

Just a curious. Again, thank you for fooocus :)

**Comment by Crocodilian12** [comment]
FWIW -- Fooocus is more of an interactive application, there's very limited batch capabilities (although there's an API)

Illyasviel also developed WebUI Forge,  which has very sophisticated batch operations like A1111  . . . you can load entire folders into slots for image prompts, Image 2 Image, and upscale. 

I'm just a user so I don't know anything about the plans for Fooocus -- and with the reversion from Mashb1t back to Illyasviel, it would be foolish to speculate. At present, it looks like the priority for Illyasviel is getting Flux and some other things running in Forge, and you'll see in the Fooocus readme:

**"The Fooocus project, built entirely on the Stable Diffusion XL architecture, is now in a state of limited long-term support (LTS) with bug fixes only. As the existing functionalities are considered as nearly free of programmartic issues (Thanks to [mashb1t](https://github.com/mashb1t)'s huge efforts), future updates will focus exclusively on addressing any bugs that may arise."**

. . . that doesn't sound like batch operations are a priority. 

me, I use Fooocus interactively. When I want to run something complicated say overnight, to produce hundreds of images, I use Forge. That's what I'd recommend to others

"It's not a big deal when it's a couple dozen, but I'm working with thousands these days for production and it's really not a joke for me anymore lol"

-- that's really more of a Forge job

https://github.com/lllyasviel/stable-diffusion-webui-forge
